frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx

```diff
@@ -188,27 +207,27 @@ export const usePipelineTaskTopology = (
       deploymentSpec: { executors },
       root: {
         dag: { tasks },
+        inputDefinitions,
       },
     } = pipelineSpec;
 
     const componentArtifactMap = parseComponentsForArtifactRelationship(components);
-    const taskArtifactMap = parseTasksForArtifactRelationship(tasks);
+    const taskArtifactMap = parseTasksForArtifactRelationship('root', tasks);
 
-    const nodes = getNodesForTasks(
-      'root',
-      spec,
-      tasks,
-      components,
-      executors,
-      componentArtifactMap,
-      taskArtifactMap,
-      runDetails,
-      executions,
-    )[0];
-
-    // Since we have artifacts that are input only that do not get created, remove any dependencies on them
-    return nodes.map((n) => ({
-      ...n,
-      runAfterTasks: n.runAfterTasks?.filter((t) => nodes.find((nextNode) => nextNode.id === t)),
-    }));
+    // There are some duplicated nodes, remove them
+    return _.uniqBy(
+      getNodesForTasks(
+        'root',
+        spec,
+        tasks,
+        components,
+        executors,
+        componentArtifactMap,
+        taskArtifactMap,
+        runDetails,
+        executions,
+        inputDefinitions?.artifacts,
+      )[0],
+      (node) => node.id,
+    );
```

Comment:
I updated this because the previous method will remove some edges from artifact nodes to task nodes.